libs/agno/agno/models/groq/groq.py

```diff
@@ -595,15 +466,17 @@ def response_stream(self, messages: List[Message]) -> Iterator[ModelResponse]:
         logger.debug("---------- Groq Response Start ----------")
         self._log_messages(messages)
         stream_data: StreamData = StreamData()
-        metrics: Metrics = Metrics()
+
+        # -*- Create assistant message
+        assistant_message = Message(role="assistant")
 
         # -*- Generate response
-        metrics.start_response_timer()
+        assistant_message.metrics.start_timer()
         for response in self.invoke_stream(messages=messages):
             if len(response.choices) > 0:
-                metrics.completion_tokens += 1
-                if metrics.completion_tokens == 1:
-                    metrics.time_to_first_token = metrics.response_timer.elapsed
+                assistant_message.metrics.completion_tokens += 1
```

Comment:
Should this not use the completion tokens from the response? I always thought this was weird, because a delta isn't necessarily one token right?